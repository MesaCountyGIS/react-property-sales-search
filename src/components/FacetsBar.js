import React from 'react';
import Datepickers from './Datepickers';
import Dropdown from './Dropdown';
import MoreFilters from './MoreFilters';


const facetsbar = {
    backgroundColor: "#eceeee",
    borderTop: "1px solid #d2d2d2",
    borderBottom: "1px solid #d2d2d2",
    boxShadow: "inset 0 6px 2px -2px #e5e5e5",
    boxSizing: "border-box",
    height: "3.2em",
    marginLeft: "-10em",
    minWidth: "890px"
}

const dividerStyle = {
    display: "inline-block",
    borderRight:"1px solid #d2d2d2",
    marginLeft: "0.5em",
    height: "1.5em"
}

// const dropdownContent = [
//   {
//    type: 'group', name: 'Improved Sales', items: [
//      { value: 'Agricultural', label: 'Agricultural' },
//      { value: 'Commercial/Industrial', label: 'Commercial/Industrial' },
//      { value: 'Condo', label: 'Condo' },
//      { value: 'Duplex/Triplex', label: 'Duplex/Triplex' },
//      { value: 'Exempt', label: 'Exempt' },
//      { value: 'Manufactured Home', label: 'Manufactured Home' },
//      { value: 'Multi Family 4 TO 8', label: 'Multi Family 4 TO 8' },
//      { value: 'Multi Family 9 AND UP', label: 'Multi Family 9 AND UP' },
//      { value: 'Single Family', label: 'Single Family' },
//      { value: 'Townhouse', label: 'Townhouse' }
//    ]
//   },
//   {
//    type: 'group', name: 'Vacant Sales', items: [
//      { value: 'Agricultural Vacant', label: 'Agricultural Vacant' },
//      { value: 'Commercial/Industrial Vacant', label: 'Commercial/Industrial Vacant' },
//      { value: 'Exempt Vacant', label: 'Exempt Vacant' },
//      { value: 'Residential Vacant', label: 'Residential Vacant' },
//    ]
//   }
// ]

const FacetsBar = () =>
  (
      <div style={facetsbar} className="facets-bar">
          <Datepickers time={4} units={'months'} />
          <span style={dividerStyle}></span>
          <Dropdown placeholder='Select Property Type' value={'Select Property Type'}/>
          <span style={dividerStyle}></span>
          <MoreFilters message={'More Filters'} />
      </div>
  );

  export default FacetsBar;
