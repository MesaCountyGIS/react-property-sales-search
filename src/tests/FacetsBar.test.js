import React from 'react';
import renderer from 'react-test-renderer';
import FacetsBar from '../components/FacetsBar';

describe("<FacetsBar />", ()=>{
    // test('renders without crashing', () => {
    //     const tree = renderer.create(
    //         <FacetsBar />
    //   ).toJSON();
    //   expect(tree).toMatchSnapshot();
    // });

    expect(`React-datepicker throws "Invariant Violation: getNodeFromInstance: Invalid argument."
    error in Jest testing when findDOMNode is used. Fork React-Datepicker to fix or file an issue`).toEqual(
        `React-datepicker throws "Invariant Violation: getNodeFromInstance: Invalid argument."
        error in Jest testing when findDOMNode is used. Fork React-Datepicker to fix or file an issue`
    )
});
