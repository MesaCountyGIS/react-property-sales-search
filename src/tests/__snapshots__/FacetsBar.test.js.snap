// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FacetsBar /> renders without crashing 1`] = `
<div
  className="facets-bar"
  style={
    Object {
      "backgroundColor": "#eceeee",
      "borderBottom": "1px solid #d2d2d2",
      "borderTop": "1px solid #d2d2d2",
      "boxShadow": "inset 0 6px 2px -2px #e5e5e5",
      "boxSizing": "border-box",
      "height": "3.2em",
      "marginLeft": "-10em",
      "minWidth": "890px",
    }
  }
>
  <span
    style={
      Object {
        "borderRight": "1px solid #d2d2d2",
        "display": "inline-block",
        "height": "1.5em",
        "marginLeft": "0.5em",
        "marginRight": "0.5em",
      }
    }
  />
  <div
    className="dropdown more-filters"
    style={Object {}}
  >
    Select Property Type
    <span
      className="dropdown more-filters"
    >
      ▼
    </span>
    <ul
      className="drop-nav display"
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    >
      <li
        className="heading"
      >
        Improved Sales
      </li>
      <li
        className="item"
      >
        Agricultural
      </li>
      <li
        className="item"
      >
        Commercial/Industrial
      </li>
      <li
        className="item"
      >
        Condo
      </li>
      <li
        className="item"
      >
        Duplex/Triplex
      </li>
      <li
        className="item"
      >
        Exempt
      </li>
      <li
        className="item"
      >
        Manufactured Home
      </li>
      <li
        className="item"
      >
        Multi Family 4 TO 8
      </li>
      <li
        className="item"
      >
        Multi Family 9 AND UP
      </li>
      <li
        className="item"
      >
        Single Family
      </li>
      <li
        className="item"
      >
        Townhouse
      </li>
      <li
        className="heading"
      >
        Vacant Sales
      </li>
      <li
        className="item"
      >
        Agricultural Vacant
      </li>
      <li
        className="item"
      >
        Commercial/Industrial Vacant
      </li>
      <li
        className="item"
      >
        Exempt Vacant
      </li>
      <li
        className="item"
      >
        Residential Vacant
      </li>
    </ul>
  </div>
  <span
    style={
      Object {
        "borderRight": "1px solid #d2d2d2",
        "display": "inline-block",
        "height": "1.5em",
        "marginLeft": "0.5em",
        "marginRight": "0.5em",
      }
    }
  />
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "display": "inline-block",
        "fontSize": "0.8em",
        "fontWeight": "bold",
        "marginTop": "1.35em",
        "position": "absolute",
      }
    }
  >
    More Filters
    <span
      style={
        Object {
          "float": "right",
          "fontSize": "0.8em",
          "marginLeft": "1em",
          "marginTop": "0.1em",
        }
      }
    >
      ▼
    </span>
    <noscript />
  </div>
</div>
`;
